### DISCUSSION TITLE
Опыт за квесты в зависимости от главы

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2nr_unofficial_update#476] [General]
Очистка статуй Инноса - Пал

Поддельный Акции - Маг

исходи из того что это квест 1-2 главы. есть там таблиц а в которой я пытался несколько раз обосновано расписать сколько опыта должно начисляться за тот или иной квест:-) 
[ ![image](https://user-images.githubusercontent.com/49514764/74594735-b30d0b80-504a-11ea-9ae3-facf5fd43bbb.png)](https://docs.google.com/spreadsheets/d/1svDbXAiYe7PLCQ7G0Arjh_njRiDabsRC-9GuLt7APsQ/edit#gid=1998166737)

**Comment by Kor-Angar** [comment]
> Возможно, лучше сделать как с Гарвеллом: чем раньше очищаем, тем больше опыта получаем.
> 
> 3 - 150
> 4 - 100
> 5 - 50

Давай. Давай сразу и акции поправим

**Comment by Kor-Angar** [comment]
еще мне очень хочется решить вопрос с опытом в Ярике.
так как разница между 2 главой в Ярике и 3 главой Ярика набегает приличная больше 1000 опыта.

- За прочтение книг в библиотеке Ярика
- За использование телепортов
- за выкапывание сокровищ
- за инфу о Вороне для Сатурасу
- еще за что то.

где то я уже писал эти моменты. этот опыт можно усреднить. давать например (100+150)/2 =125 опыта. Либо давать за одни моменты 100 опыта, а за другие 150.  

мне лично как игроку непринципиально как оно будет так как я всегда ходил во 2 главе в ярик( у нас так принято было с 2004 года было). а в ярик стал в 3 главе ходить в рамках этого обновления что бы в логике авторов разобраться :-)

**Comment by Kor-Angar** [comment]
разобраться в логике авторов не смог, но нашел ответ в затраченном времени в главах. если Ярик весь пройти во 2 главе, то 3 глава получается очень коротка. а если весь ярик перенести на 3 главу, то 2 глава получается маленькая. видимо авторы предполагали что в какой то момент ярик может стать сложным и тогда герой уйдёт в  3 главу по основному сюжету и в 3 главе сможет доиграть Ярик в доспехах Пала/ОНД.
тут в принципе все логично так как для первого прохождения без знания всех бонусов, нычек, оружия и прочего, Ярик может быть достаточно сложен.

**Comment by dosinabox** [comment]
Хотелось сделать с Джо как с инфой для Гарвелла (чем раньше найдем, тем больше опыта получаем), но получается как-то слишком для первой главы.

Сейчас:
1 - 200
2 - 400
3 - 600
4 - 800
5 - 1000

Хочется:
1 - 1000
2 - 800
3 - 600
4 - 400
5 - 200

**Comment by Kor-Angar** [comment]
исходи из того что это квест 1-2 главы. есть там таблиц а в которой я пытался несколько раз обосновано расписать сколько опыта должно начисляться за тот или иной квест:-) 
[ ![image](https://user-images.githubusercontent.com/49514764/74594735-b30d0b80-504a-11ea-9ae3-facf5fd43bbb.png)](https://docs.google.com/spreadsheets/d/1svDbXAiYe7PLCQ7G0Arjh_njRiDabsRC-9GuLt7APsQ/edit#gid=1998166737)
исходил из такой логики 1 глава - 200, 2 глава - 400, значит (200+400)/2 = 300. 300 можно просто сделать фиксированным. нашёл в первой главе вместо 200 получил 300. нашёл во второй вместо 400 получил 300 :-) нашёл позже, всё равно получил 300.  

Кстати, про Джо рассказывает торговец на рыночной площаде. он тоже даёт опыт, если у него спросить о пропавших
Сейчас:
- 1 - 50
- 2 - 100
- 3 - 150
- 4 - 200
- 5 - 250   

Но это же вовсе не значит что разговор с ним в 1 главе надо давать 250 опыта. или значит?:-)  его бы зафиксировать в размере 50 опыта. ну или 75 :-)

всё сводится к тмоу какая мысль была авторов игры. Я так понял что это было сделано из экономии времени. если это так то всё достаточно просто, если квест 1 главы то фиксируется опыт соответствующей первой главе. если какой-то квест спорный, то ли 2 главы, то ли 3 главы, то среднеарифметическое. и вот таким образом ручками его просто фиксировать. но это мне так видится. 

заметь что самая простая история с квестами 5 главы, там вообще никаких вариантов нет :-) или с НПС которые есть только во 2 главе, например Мильтен, Горн, Диего в Минентале :-) или Горн наём в 3 главе в Хориносе. или Горн ОНД в 4 главе в Минентале. если ориентироваться на них становится намного понятнее сколько и за что давать опыта :-)

**Comment by Kor-Angar** [comment]
таким образом, если этот вопрос пытаться прорабатывать то лучше начать с  конца, т.е. с квестов 4 главы  :-)

**Comment by Kor-Angar** [comment]
опыт за пожертвования Дарона тоже привязан к Главе? :-)
```
	else if((Daron_Spende < 500) && (Bonus_2 == FALSE))
	{
		B_GivePlayerXP(XP_Ambient);
		Bonus_2 = TRUE;
	}
```

**Comment by dosinabox** [comment]
> опыт за пожертвования Дарона тоже привязан к Главе? :-)

Получается, так.

> исходи из того что это квест 1-2 главы.

Ну да. И не хочется, чтобы игроки откладывали его на пятую, чтобы получить больше опыта. Сейчас получается так, что игрока поощряют за то, что он забивает на поиски Джо. А человек может быть в опасности.

С Гарвеллом такое изменение прокатило, потому что там разница в опыте минимальна. А тут разбег аж от 200 до 1000 - или теряешь 800 в пятой, или получаешь 800 в первой (что очень много для этой стадии).

**Comment by dosinabox** [comment]
> А тут разбег аж от 200 до 1000

Возможное решение:

1 - 200 -> 600
2 - 400 -> 550
3 - 600 -> 500
4 - 800 -> 450
5 - 1000 -> 400

**Comment by Kor-Angar** [comment]
> Возможное решение:

плохо решение. не понимаю зачем ты пытаешь в первой главе накачать героя опытом холявным.

авторы игры не подразумевали что герой за  этот квест поучит 1000 опыта. возможность получить за этот квест столько опыта это  следствие работы скрипта.

если мы говорим о авторской логике, то получаемое кол-во опыта привязано к главе. за квесты 1 главы 50 опыта, 2 глава 100 опыта и так далее.

получение большего кол-во опыта или меньшего кол-ва опыта за один и тот же квест это плохая история в целом.

**Comment by Kor-Angar** [comment]
Джо это квест 1-2 главы. опыт за Джо это 200 или 400. среднеарифметическое это 300.

и возможно стоит подумать о том что бы сделать невозможным выполнения это квеста в поздних главах. не может же Джо сидеть там до 4-5 главы :-)

**Comment by Kor-Angar** [comment]
Давай возьмем два одинаковых простых квеста. например про то что у рыбака забирают рыбу и про чувака который стучит по дому.

оба эти квесты доступны с 1 главы. НО за один получаемый опыт привязан к главе игры, а другой нет. 

вопрос: чем они отличаются? в чём разницу? где ошибка? ;-)

**Comment by Kor-Angar** [comment]
const int XP_Addon_FarimSchutz = 50;
const int XP_Addon_Farim_PaladinFisch = 100;

**Comment by dosinabox** [comment]
> и возможно стоит подумать о том что бы сделать невозможным выполнения это квеста в поздних главах. не может же Джо сидеть там до 4-5 главы :-)

Удалять Джо из мира и добавлять в башню модель скелетика?)

В общем, конкретно за Джо ты предлагаешь фиксированные 300 опыта независимо от главы?

**Comment by Kor-Angar** [comment]
> В общем, конкретно за Джо ты предлагаешь фиксированные 300 опыта независимо от главы?

да.я так же предлагаю это и для большинства других квестов :-)
вот даже в позапрошлом году таблицу начал делать
https://docs.google.com/spreadsheets/d/1svDbXAiYe7PLCQ7G0Arjh_njRiDabsRC-9GuLt7APsQ/edit#gid=1998166737

> Удалять Джо из мира и добавлять в башню модель скелетика?)

я бы просто в портовую таверну отправлял его в 3 главе, после освобождения Беннета, так как его  мог кто то и другой выпустить. в конце концов он член организации и они бы наверняка его нашли :-)

**Comment by dosinabox** [comment]
> вот даже в позапрошлом году таблицу начал делать

Знаю, но Джо там нет.

**Comment by dosinabox** [comment]
> я бы просто в портовую таверну отправлял его в 3 главе

Или в тюрьму, потому что его могли найти не воры, а ополчение)

**Comment by Kor-Angar** [comment]
> Или в тюрьму, потому что его могли найти не воры, а ополчение)

Логично. тогда с начало третей главы, в этом момент Пек идёт в казарму ополчения из Барделя и у него как раз ключ от Башни в которой сидит Джо :-)

**Comment by Kor-Angar** [comment]
> Знаю, но Джо там нет.

нету. там что то для начало было набросано, что бы обсудить это и выработать некие правила :-)

**Comment by Kor-Angar** [comment]
я бы точечно начал правки делать по опыту. с 4 главу. и с опыта за квесты в Ярике.

**Comment by Kor-Angar** [comment]
ну **вот хотя бы** для начало в Ярике опыт за

- использование телепортов
- прочтение книг
- выкапывание кладов

я во второй главе когда опыт получаю за это дело прям напрягаюсь, что в 3 главе дают больше :-)

**Comment by Kor-Angar** [comment]
Квесты который можно выполнить в первой главе:

![image](https://user-images.githubusercontent.com/49514764/93693161-ae3e9780-fb04-11ea-9c14-b8f3df8241fc.png)
![image](https://user-images.githubusercontent.com/49514764/93693132-6455b180-fb04-11ea-92c3-1f1633ac6e14.png)
![image](https://user-images.githubusercontent.com/49514764/93693120-37090380-fb04-11ea-8795-a84595ee3e1d.png)
![image](https://user-images.githubusercontent.com/49514764/93693092-d2e63f80-fb03-11ea-96a7-93b2c5316410.png)

**Comment by Kor-Angar** [comment]
квесты которые не выполняет нейтрал в 1 главе
![image](https://user-images.githubusercontent.com/49514764/93693173-ce6e5680-fb04-11ea-8e65-555e5d502a2f.png)

**Reply by Kor-Angar** [reply]
если не выполняет нейтрал то из этого вовсе не следует что их не сможет выполнить герой вступивший во фракцию ;-)

**Comment by Kor-Angar** [comment]
можно поменять  XP_Ambient в файлах

evt_teleportstation_func.d - XP_AmbientKap2
Bookstand_Addon.d - XP_AmbientKap2
DIA_SLD_805_Cord.d -  XP_AmbientKap2
DIA_VLK_422_Salandril.d - XP_AmbientKap3
DIA_VLK_409_Zuris.d - XP_AmbientKap3
DIA_VLK_404_Lutero.d - XP_Ambient3
DIA_VLK_424_Alwin.d - XP_AmbientKap1
DIA_VLK_417_Constantino.d - XP_AmbientKap1
DIA_MIL_337_Mika.d  -  XP_AmbientKap2
DIA_PIR_1301_Addon_Skip_NW.d  -  XP_AmbientKap2



***
DIA_VLK_407_Hakon.d
AI_Output(self,other,"DIA_Hakon_Minenanteil_12_01");	//Извини, но я не могу сказать тебе этого. Это слишком опасно для меня.
	B_GivePlayerXP(XP_AmbientKap3);
***
DIA_NOV_612_Babo.d
(XP_Ambient * 2)  на  (XP_AmbientKap1* 2)
(XP_Ambient) на (XP_AmbientKap3) 
***


https://github.com/Kor-Angar/G2_NotR_Kor_Angar/commit/1ccf818070847119d7d91b569e67f689551eeb0b

**Comment by dosinabox** [comment]
> DIA_VLK_407_Hakon.d
> AI_Output(self,other,"DIA_Hakon_Minenanteil_12_01"); //Извини, но я не могу сказать тебе этого. Это слишком опасно для меня.
> B_GivePlayerXP(XP_AmbientKap3);

Разве "Поддельные акции" нельзя выполнить в 4 главе?


> DIA_NOV_612_Babo.d
> (XP_Ambient * 2) на (XP_AmbientKap1* 2)

Можно получить хоть в 5 главе.

> (XP_Ambient) на (XP_AmbientKap3)

Доступно в 3, 4 и 5 главах.

**Comment by Kor-Angar** [comment]
всё верно, сейчас квест из 1 главы можно как взять так и выполнить в 5 главы и получить 250 опыта так как  переменная динамичная XP_Ambient.

а это тема как раз об обратном.  о том что если квест первой главы, то и опыт за него не должен быть динамический. 

поддельные акции квест 3 главы, получается "автоматически"
задания про Бабо квест 1 главы, доступный только Послушнику
и так далее.

если это править, то править исходя из какой то логики. в прошлом году я делал таблицу в который отмечал в какой главе квест берется, в какой его можно сдать, насколько квест сложен / важен и сомневаясь вручную прописывал опыт. в этом году я выполнил все задания которые можно выполнить в 1 главе, не убивая всех монстров, посмотрел на уровень, оценил баланс и пришёл к выводы что такая ювелирная работа правильна по смыслу но бессмысленная с точки зрения практики и не будет никогда реализована. 
в связи с этим я решил расставить всё по нижней планке, если квест из 1 главы то XP_AmbientKap1 из второй XP_AmbientKap2 и так далее. а потом на уровне скриптов сделать переключалку:
в одном случае XP_AmbientKap1  = 50, а в другом XP_AmbientKap1  = XP_Ambient

таким образом будет возможность получать опыт как в оригинальной игре так и  включить честный расчёт опыта. я это в своей сборке реализуя и потестирую.

**Comment by Kor-Angar** [comment]
даже так? :-)
https://github.com/dosinabox/g2nr_unofficial_update/blob/aed56ebb175bc52740bf908b2ece4ef16490b17d/PrjGOTHIC/Story/Dialoge/DIA_NONE_100_Xardas.d#L581

**Comment by Kor-Angar** [comment]
ну тут точно только 4 глава :-)
![image](https://user-images.githubusercontent.com/49514764/94353334-c1a7b080-0078-11eb-962a-e870e8bb662c.png)

**Comment by Kor-Angar** [comment]
первые правки
https://github.com/Kor-Angar/G2_NotR_Kor_Angar/commit/1ccf818070847119d7d91b569e67f689551eeb0b

второй массив правок
https://github.com/Kor-Angar/G2_NotR_Kor_Angar/commit/e811397267b646db69af8ec1278267a13bc3e370

третья порция правок (28,09,2020)
https://github.com/Kor-Angar/G2_NotR_Kor_Angar/commit/2cc56e5b08b62557cfe6bf62f28a622bc91e407d

теперь вроде всё, XP_Ambient был побеждён как класс :-) остались только  которые стали динамическими :-)
```
XP_AmbientKap1;
XP_AmbientKap2;
XP_AmbientKap3;
XP_AmbientKap4;
XP_AmbientKap5;
XP_AmbientKap6;
```
есть так же  возможность включить / выключить динамическое начисление опыта, 
https://github.com/Kor-Angar/G2_NotR_Kor_Angar/commit/09d4f4077c683da5dcf28b04c51656d087944cf3
т.е. за квест 1 главы XP_AmbientKap1 (например разговор с Ксардасом о табличке) [можно получить 150 опыта в 3 главе](https://www.youtube.com/watch?v=5UsuDMJdrtI). или за поддельные акции XP_AmbientKap3 получить 250 опыта в 5 главе и т.д.

в практическом плане это хорошо для каких то игроков которых чувство жадности заставляло переносить сдачу квестов на поздние главы. в принципе эту задачу оно и должно было решать в конечном счёте :-)
возможно конечно где то в правках мною было допущены ошибка и квест 1 главы я пометил неосознанно XP_AmbientKap2 но это человеческий фактор, такое можно выявить/исключить только коллективным трудом.

**Comment by Kor-Angar** [comment]
единственно что не получилось мне разделить за нычки Грега в Хориносе (первая глава) и Ярике (вторая-третья главы). можно было что то сочинить, но я решил что это излишне и просто сделал XP_AmbientKap2.

upd 
так и оказалось :-) Мика
https://github.com/Kor-Angar/G2_NotR_Kor_Angar/commit/0872cb769ca025f500b61a32ccb5446f143da748

**Comment by Kor-Angar** [comment]
Прошёл так игру  не сколько раз. Никакого дисбаланса не обнаружил.

**Comment by Kor-Angar** [comment]
как быстро время летит.. в честь моего завтрашнего похода на ковёр я решил убить вечер и использовать новый функционал zparser что бы иметь возможность вынести данный проект в мини мод. новая вкладка таблички 
![image](https://user-images.githubusercontent.com/49514764/202579211-6afcca86-f43b-46c6-97b5-b0f87e310f87.png) https://docs.google.com/spreadsheets/d/1svDbXAiYe7PLCQ7G0Arjh_njRiDabsRC-9GuLt7APsQ/edit#gid=2024314332

в самой табличке две основные колонки: это название инстанции диалога и минимальное значение главы в котором  можно сдать квест.
![image](https://user-images.githubusercontent.com/49514764/202579554-a0bd5ca3-80c8-4c37-8d1d-2a06c2c62b85.png)

далее в `B_GivePlayerXP ` идёт проверка и корректируется получаемы опыт
```    if (C_DIA_CurrentName("DIA_Mika_HILFE"))     { add_xp = XP_AmbientKap1; }; ```

**Reply by Kor-Angar** [reply]
```D
var int DIA_CurrentInstance ;
var string DIA_CurrentName;

func int C_DIA_CurrentName(var string instmisname)
{
    if (Hlp_StrCmp(DIA_CurrentName,instmisname))
    {
        return true;
    };
    return false;
};
```

**Reply by Kor-Angar** [reply]
https://steamcommunity.com/sharedfiles/filedetails/?id=2889895105

**Reply by Kor-Angar** [reply]
раскосые файлики 
[KA_G2_GivePlayerXP.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/10037028/KA_G2_GivePlayerXP.zip)

**Reply by Kor-Angar** [reply]
закомментировал

- DIA_Lee_RichterBeweise
- DIA_Buster_BringTrophyShadowbeast
- DIA_Vatras_BEGINN


[B_GivePlayerXP.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/10049022/B_GivePlayerXP.zip)

**Comment by Kor-Angar** [comment]
@dosinabox помоги пожалуйста

```D
func void :B_GivePlayerXP (var int add_xp)
{
    // квесты первой главы
    ...
    //  у Ярвиса опыт начисляется в зависимости от кол-ва побед
    //  B_GivePlayerXP(XP_Ambient * victories);
    else if (C_DIA_CurrentName("DIA_Jarvis_HowManyLeft"))                    { add_xp = (add_xp / Kapitel * 1) ; }
```
вот так если написать то это будет тоже самое как если бы `XP_AmbientKap1 * victories` ?

**Reply by dosinabox** [reply]
покажи аргументы функции B_GivePlayerXP в DIA_Jarvis_HowManyLeft_Info

**Reply by Kor-Angar** [reply]
//  B_GivePlayerXP(XP_Ambient * victories);

**Reply by dosinabox** [reply]
И в чем твоя задумка? Пока не понимаю ни вопрос, ни код.

**Reply by Kor-Angar** [reply]
1) у меня есть мини-мод. он перебивает получаемый опыт
2) Ярвис начисляет опыт привязанные к количествам побед B_GivePlayerXP(XP_Ambient * victories)
3) я хочу XP_Ambient * victories изменить на XP_AmbientKap1 * victories
4) для этого в B_GivePlayerXP  я беру add_xp = XP_Ambient * victorie и преобразовываю в   add_xp = (add_xp / Kapitel * 1) ;

**Reply by dosinabox** [reply]
А что здесь есть 1?

**Reply by Kor-Angar** [reply]
здесь 1 что бы  я понимал что опыта как в 1 главе. ниже я использую такую же конструкцию и использую цифру 2
 ```D
   // ГЛАВАВ 1

    else if (C_DIA_CurrentName("DIA_Jarvis_HowManyLeft"))                    { add_xp = (add_xp / Kapitel * 1) ; }
    
    // ГЛАВАВ 2    
    
    else if (C_DIA_CurrentName("DIA_Addon_Pardos_trank_Info"))           { add_xp = (add_xp / Kapitel * 2); }
```

**Reply by Kor-Angar** [reply]
допустим 1 глава, герой победил 3
B_GivePlayerXP (50*3)

add_xp = 50 * 3 / 1 * 1 = 150

**Reply by Kor-Angar** [reply]
допустим 3 глава, герой победил 3
B_GivePlayerXP (150*3)

add_xp = 150 * 3 / 3 * 1 = 150

**Reply by dosinabox** [reply]
Чтобы получить `XP_AmbientKap1 * victories`, в B_GivePlayerXP нужно поделить входящий опыт от Джарвиса на текущую главу:
```cpp
add_xp /= Kapitel;
```

**Reply by Kor-Angar** [reply]
ну вот и мне так кажется

**Comment by Kor-Angar** [comment]
в диалогах `DIA_Abuyin_Zukunft***_Trance` можно вынести анимацию, эффект `B_GivePlayerXP` и `B_RemoveEveryInvItem`  в отдельную общую функцию

```D
	AI_PlayAni(self,"T_MAGRUN_2_HEASHOOT");
	Wld_PlayEffect("SPELLFX_TELEPORT",self,self,0,0,0,FALSE);
	B_GivePlayerXP(XP_Ambient * 4);
	B_RemoveEveryInvItem(self,ItMi_Gold);
```

**Reply by dosinabox** [reply]
```cpp
func void B_Abuyin_StartTrance()
{
	AI_PlayAni(self,"T_MAGRUN_2_HEASHOOT");
	Wld_PlayEffect("SPELLFX_TELEPORT",self,self,0,0,0,FALSE);
	Abuyin_Zukunft = Kapitel;
	B_GivePlayerXP(XP_Ambient * 4);
	B_RemoveEveryInvItem(self,ItMi_Gold);
};
```

**Reply by Kor-Angar** [reply]
кинь ссылки плиз на правку, как добавишь

**Reply by dosinabox** [reply]
https://github.com/dosinabox/g2nr_unofficial_update/commit/e1cd4dfc41e57aa662683349a77c31fb80c22828

**Comment by Kor-Angar** [comment]
**XP_AmbientKap5**

**Reply by Kor-Angar** [reply]
```D
func void DIA_Buster_DealerIsDead_Info()
{
	AI_Output(self,other,"DIA_Buster_BringTrophyShadowbeast_13_00");	//Мой торговец из города передал прощальный привет.
	AI_Output(other,self,"DIA_Buster_BringTrophyShadowbeast_15_01");	//Что это значит?
	AI_Output(self,other,"DIA_Buster_BringTrophyShadowbeast_13_02");	//Он мертв. Можешь оставить эти рога себе. Теперь я все равно не знаю, что с ними делать.
	MIS_Buster_KillShadowbeasts_DJG = LOG_SUCCESS;
	if((BusterTrophyCounter > 0) || Npc_HasItems(other,ItAt_ShadowHorn))
	{
		B_GivePlayerXP(XP_Ambient );
//		B_GivePlayerXP(XP_AmbientKap5);
	}
	else
	{
		B_CheckLog();
	};
	AI_StopProcessInfos(self);
};

```

**Reply by Kor-Angar** [reply]
```D
func void DIA_Torlof_BEMYCAPTAIN2_Info()
{
	AI_Output(other,self,"DIA_Torlof_BEMYCAPTAIN2_15_00");	//Ворота замка в Долине Рудников открыты и их заклинило. Теперь ничто не сдерживает орков, и они берут этот замок приступом.
	AI_Output(other,self,"DIA_Torlof_BEMYCAPTAIN2_15_01");	//Паладины в замке несут тяжелые потери.
	AI_Output(other,self,"DIA_Torlof_BEMYCAPTAIN2_15_02");	//И, я думаю, не пройдет много времени, когда паладины из города выступят в Долину Рудников, чтобы выручить парней, осажденных в замке.
	AI_Output(self,other,"DIA_Torlof_BEMYCAPTAIN2_01_03");	//Это хорошие новости. Теперь ничто не мешает мне покинуть это богом проклятое место.
	B_GivePlayerXP(XP_Ambient );
	//B_GivePlayerXP(XP_AmbientKap5);
};
```

**Reply by Kor-Angar** [reply]
```D
func void DIA_Pyrokar_IRDORATHBOOKOPEN_glueck()
{
	AI_Output(other,self,"DIA_Pyrokar_IRDORATHBOOKOPEN_glueck_15_00");	//Просто слепая удача, я думаю.
	AI_Output(self,other,"DIA_Pyrokar_IRDORATHBOOKOPEN_glueck_11_01");	//Не говори чепухи. Удача!
	AI_Output(self,other,"DIA_Pyrokar_IRDORATHBOOKOPEN_glueck_11_02");	//Если даже я не смог открыть эту книгу, а затем приходит кто-то вроде тебя и играючи открывает ее...
	AI_Output(self,other,"DIA_Pyrokar_IRDORATHBOOKOPEN_glueck_11_03");	//... это наводит меня на грустные размышления...
	AI_Output(self,other,"DIA_Pyrokar_IRDORATHBOOKOPEN_glueck_11_04");	//Ладно. Так как ты, очевидно, единственный, кто смог открыть эту книгу, то я, пожалуй, позволю тебе держать ее у себя. По крайней мере, пока мы не разрешили этот кризис.
	//B_GivePlayerXP(XP_AmbientKap5);
	B_GivePlayerXP(XP_Ambient);
	Info_ClearChoices(DIA_Pyrokar_IRDORATHBOOKOPEN);
};
```

**Reply by Kor-Angar** [reply]
```D
func void DIA_Rosi_FLEEFROMSEKOB_Info()
{
	AI_Output(other,self,"DIA_Rosi_FLEEFROMSEKOB_15_00");	//Что ты делаешь здесь, в глуши?
	AI_Output(self,other,"DIA_Rosi_FLEEFROMSEKOB_17_01");	//Я больше не в силах терпеть все это. Секоб начал закатывать истерики по любому поводу.
	if(RosiToldAboutSekob == FALSE)
	{
		B_RosiAboutSekob();
	};
	AI_Output(self,other,"DIA_Rosi_FLEEFROMSEKOB_17_02");	//В конце концов, все перестали с ним разговаривать. Ну и он тоже не говорит ни с кем.
	AI_Output(self,other,"DIA_Rosi_FLEEFROMSEKOB_17_03");	//Я должна была бежать оттуда, но теперь я не знаю, куда мне идти.
	AI_Output(self,other,"DIA_Rosi_FLEEFROMSEKOB_17_04");	//Ты не мог бы помочь мне?
//	B_GivePlayerXP(XP_AmbientKap5);
	B_GivePlayerXP(XP_Ambient);
	RosiFoundKap5 = TRUE;
};
```

**Reply by Kor-Angar** [reply]
```D
func void DIA_Sekob_ROSIBACKATSEKOB_Info()
{
	AI_Output(other,self,"DIA_Sekob_ROSIBACKATSEKOB_15_00");	//Я привел твою жену назад.
	AI_Output(self,other,"DIA_Sekob_ROSIBACKATSEKOB_01_01");	//Превосходно. Вот, возьми это за свои труды.
	MIS_bringRosiBackToSekob = LOG_SUCCESS;
	CreateInvItems(self,ItMi_Gold,650);
	B_GiveInvItems(self,other,ItMi_Gold,650);
//	B_GivePlayerXP(XP_AmbientKap5);
	B_GivePlayerXP(XP_Ambient);
	B_NpcClearObsessionByDMT(self);
	Rosi.aivar[AIV_PARTYMEMBER] = FALSE;
	B_StartOtherRoutine(Rosi,"Start");
	if(!Npc_IsDead(Till))
	{
		Till.aivar[AIV_PARTYMEMBER] = FALSE;
		B_StartOtherRoutine(Till,"Start");
	};
};

```

**Reply by Kor-Angar** [reply]
```D
func void DIA_Lee_DRACHENEI_Info()
{
	AI_Output(other,self,"DIA_Lee_DRACHENEI_15_00");	//Люди-ящеры раскладывают драконьи яйца по всему острову.
	AI_Output(self,other,"DIA_Lee_DRACHENEI_04_01");	//Я мог понять это раньше. Пришло время убираться отсюда.
	if(hero.guild == GIL_DJG)
	{
		AI_Output(other,self,"DIA_Lee_DRACHENEI_15_02");	//А что мне делать с ними?
		AI_Output(self,other,"DIA_Lee_DRACHENEI_04_03");	//Разбей их. Что еще?
		AI_Output(self,other,"DIA_Lee_DRACHENEI_04_04");	//Может быть, из скорлупы можно будет сделать доспехи или еще что-нибудь.
		AI_Output(self,other,"DIA_Lee_DRACHENEI_04_05");	//Похоже, они очень крепкие. Поговори об этом с Беннетом.
		if(DRACHENEIER_angebotenXP_OneTime == FALSE)
		{
			if(TOPIC_END_DRACHENEIER == FALSE)
			{
				Log_CreateTopic(TOPIC_DRACHENEIER,LOG_MISSION);
				Log_SetTopicStatus(TOPIC_DRACHENEIER,LOG_Running);
			};
			B_LogEntry(TOPIC_DRACHENEIER,"Ли не знает, что делать с драконьим яйцом. Он отправил меня к кузнецу Беннету.");
		};
	};
	//B_GivePlayerXP(XP_AmbientKap5);
	B_GivePlayerXP(XP_Ambient);
};
```
